### ISSUE TITLE
Prune service log buffers

### ISSUE THREAD

**Issue by benhoyt** [canonical/pebble#539]
If you've started lots of services and then stopped them, the memory used by the service information (particularly the log ring buffer) stays around and is never pruned. Pebble is used in several contexts including memory-constrained containers and small devices, so we should design and implement a way to prune these when the services has been stopped for a while or there are too many services. Obviously we can't prune running services; we may be able to shrink/deallocate the log buffers for services on backoff state if they've been in that state for too long.

For comparison, old (done) changes are pruned when there are more than 500 of them or they've been done for more than 7 days (the interval for checking whether to prune is 10m).

**Comment by benhoyt** [comment]
@M7mdisk No, I believe that error is due to https://github.com/canonical/pebble/issues/376, which was fixed in #384 (Pebble [v1.10.0](https://github.com/canonical/pebble/releases/tag/v1.10.0)). I'll comment on that issue.

**Comment by IronCore864** [comment]
At the sprint, we discussed it, and we decided to:

- not do anything about checks because each item is only about 100 bytes.
- delete serviceData that is inactive if it's older than 7 days, or more than 1000 inactive items, sorted by currentSince.
